src/CSnakes.Runtime/CPython/Coroutine.cs

```diff
@@ -0,0 +1,82 @@
+﻿using CSnakes.Runtime.Python;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+
+namespace CSnakes.Runtime.CPython;
+internal unsafe partial class CPythonAPI
+{
+    internal static bool IsPyCoroutine(PyObject p)
+    {
+        return HasAttr(p, "__await__");
+    }
+
+    internal class EventLoop : IDisposable
+    {
+        private readonly PyObject? loop = null;
+
+        public EventLoop()
+        {
+            loop = NewEventLoopFactory!.Call();
+        }
+
+        public bool IsDisposed { get; private set; }
+
+        private void Close()
+        {
+            if (loop is null)
+            {
+                throw new InvalidOperationException("Event loop not initialized");
+            }
+            using var close = loop.GetAttr("close");
+            close?.Call();
+        }
+
+        public void Dispose()

```

Comment:
We should ensure that `Dispose()` can be called multiple times. Not sure if this implementation handles that.